packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3235 @@
+  ///
+  /// The color in this text style will only be used if [buttonOverlayColor]
+  /// is unset.
+  final MaterialStateProperty<TextStyle?>? buttonTextStyle;
+
+  /// Called when the button that opens the submenu is hovered over.
+  @override
+  final ValueChanged<bool>? onHover;
+
+  @override
+  final VoidCallback? onOpen;
+
+  @override
+  final VoidCallback? onClose;
+
+  @override
+  final List<MenuItem> menus;
+
+  @override
+  State<MenuBarMenu> createState() => _MenuBarMenuState();
+
+  @override
+  Iterable<Map<String, Object?>> toChannelRepresentation(PlatformMenuDelegate delegate,
+      {required int Function(MenuItem) getId}) {
+    return <Map<String, Object?>>[PlatformMenu.serialize(this, delegate, getId)];
+  }
+
+  @override
+  List<MenuItem> get descendants => PlatformMenu.getDescendants(this);
+
+  @override
+  List<DiagnosticsNode> debugDescribeChildren() {
+    return <DiagnosticsNode>[
+      ...menus.map<DiagnosticsNode>((MenuItem child) {
+        return child.toDiagnosticsNode();
+      })
+    ];
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<Widget>('leadingIcon', leadingIcon, defaultValue: null));
+    properties.add(StringProperty('label', label));
+    properties.add(DiagnosticsProperty<Widget>('trailingIcon', trailingIcon, defaultValue: null));
+    properties.add(StringProperty('semanticLabel', semanticLabel, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<Color?>>('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<ShapeBorder?>>('shape', shape, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<double?>>('elevation', elevation, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('padding', padding, defaultValue: null));
+    properties.add(DiagnosticsProperty<EdgeInsets?>('buttonPadding', buttonPadding, defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonBackgroundColor', buttonBackgroundColor,
+        defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonForegroundColor', buttonForegroundColor,
+        defaultValue: null));
+    properties.add(DiagnosticsProperty<MaterialStateProperty<Color?>>('buttonOverlayColor', buttonOverlayColor,
+        defaultValue: null));
+    properties
+        .add(DiagnosticsProperty<MaterialStateProperty<ShapeBorder?>>('buttonShape', buttonShape, defaultValue: null));
+    properties.add(
+        DiagnosticsProperty<MaterialStateProperty<TextStyle?>>('buttonTextStyle', buttonTextStyle, defaultValue: null));
+  }
+}
+
+class _MenuBarMenuState extends State<MenuBarMenu> {
+  _MenuNode? menu;
+  _MenuBarController? controller;
+  bool registered = false;
```

Comment:
This property is unused, I believe?

(here and for other classes as well, the properties should be made private so the analyzer can tell us when a property is unused. It doesn't do that for public properties even in private classes, as far as I can tell.)

Reply:
`hoverTimer` below also seems unused.

Reply:
Correct!  I converted them all to private variables so that the analyzer could help here.